bokehjs/src/lib/models/tools/button_tool.ts

```diff
@@ -20,7 +22,13 @@ export abstract class ButtonToolButtonView extends DOMView {
 
   render(): void {
     empty(this.el)
-    this.el.classList.add(this.model.icon)
+    const icon = this.model.icon
+    if (isString(icon)) {
+      if (startsWith(icon, "bk-tool-icon"))
+        this.el.classList.add(icon)
+      else
+        this.el.style.backgroundImage = "url('data:image/png;base64," + icon + "')"
```

Comment:
I would change (reverse) this logic, especially in light of #7499. Anything that starts with `data:` should be treated as a CSS icon and everything else should be a CSS class. This also means that `icon` should contain `data:image/png;base64,` prefix. This way we would allow for other formats and encodings.